crates/fuzzing/wasm-spec-interpreter/ocaml/interpret.ml

```diff
@@ -55,7 +55,7 @@ let extract_exported_func export = match export with
 (** Interpret the first exported function and return the result. Use provided
 parameters if they exist, otherwise use default (zeroed) values. *)
 let interpret_exn module_bytes opt_params =
-  let opt_params_ = Option.map (List.map convert_to_wasm) opt_params in
+  let opt_params_ = Option.map (List.rev_map convert_to_wasm) opt_params in
```

Comment:
Is this a bug in the official wasm interpreter?

Reply:
Internally, the interpreter represents the value stack as a list in reverse order, so that popping from the stack is equivalent to taking the head of the list.

My tweaked version of the interpreter exposes an entrypoint function that assumes this reversal has already happened, hence the confusion. When I cut a new version of my interpreter, I can investigate if it makes sense to push the reversal down further, but an advantage of doing it this way is that the `rev` and `map` operations can be carried out simultaneously, which is more efficient.